cmd/clusterctl/internal/util/objs.go

```diff
@@ -105,42 +120,87 @@ func FixImages(objs []unstructured.Unstructured, alterImageFunc func(image strin
 	// look for resources of kind Deployment and alter the image
 	for i := range objs {
 		o := &objs[i]
-		if o.GetKind() != deploymentKind {
-			continue
+		err := fixDeploymentImages(o, alterImageFunc)
+		if err != nil {
+			return nil, err
 		}
-
-		// Convert Unstructured into a typed object
-		d := &appsv1.Deployment{}
-		if err := scheme.Scheme.Convert(o, d, nil); err != nil {
+		err = fixDaemonSetImages(o, alterImageFunc)
+		if err != nil {
 			return nil, err
 		}
+		objs[i] = *o
+	}
+	return objs, nil
```

Comment:
Do we need to return `objs` here? Seems like we're doing in place updates of the slice?

Reply:
I think that this is not required but it's part of the function interface!

Reply:
cc @fabriziopandini We might want to revisit ^